src/Dashboard/Http/Controllers/AuthenticateDashboard.php

```diff
@@ -2,13 +2,27 @@
 
 namespace BeyondCode\LaravelWebSockets\Dashboard\Http\Controllers;
 
+use Pusher\Pusher;
 use Illuminate\Http\Request;
+use BeyondCode\LaravelWebSockets\Apps\App;
 use Illuminate\Contracts\Broadcasting\Broadcaster;
 
 class AuthenticateDashboard
 {
     public function __invoke(Request $request, Broadcaster $broadcaster)
     {
+        /**
+         * Find the app by using the header
+         * and then reconstruct the PusherBroadcaster
+         * using our own app selection.
+         */
+        $app = App::findById($request->header('x-app-id'));
+
+        $broadcaster->__construct(new Pusher(
```

Comment:
Please don't call the constructor method, but instead create a new Broadcaster instance using the `new` keyword. We can then remove the Broadcaster in the method declaration.